### PR TITLE
Add `@CanIgnoreReturnValue` to generated Validating Builders

### PR DESCRIPTION
**Pull request by dmitrykuzmin**
Previously, the generated `ValidatingBuilder` descendants fell under the `@CheckReturnValue` rule and the return values of their methods could not be safely ignored. This caused inconveniences with their usage.

Now, `ValidatingBuilder` setter methods of various forms will be annotated with `@CanIgnoreReturnValue`.

List of methods which are affected by `@CanIgnoreReturnValue`:
1. `set...`
2. `setRaw...`
3. `clear...`
4. `addAll...`
5. `remove...`
6. `addRaw...`
7. `addAllRaw`
8. `put...`
9. `putAll...`
10. `putRaw...`
11. `putAllRaw...`

Basically, all methods have `@CanIgnoreReturnValue` except static `newBuilder()` and getters.

Spine version advances to `0.10.48-SNAPSHOT`.

### CODE REVIEW

**Comment by dmitrykuzmin** [comment]
@armiol PTAL.

File: tools/model-compiler/src/main/java/io/spine/tools/compiler/validation/ValidatingBuilderWriter.java [hunk 1]
```diff
@@ -87,6 +88,7 @@ void write(VBType type) {
                                    messageBuilderClassName,
                                    methodsAssembler.createMethods())
                         .addAnnotation(generatedBySpineModelCompiler())
+                        .addAnnotation(canIgnoreReturnValue())
```
Comment (armiol) [inline tools/model-compiler/src/main/java/io/spine/tools/compiler/validation/ValidatingBuilderWriter.java]: There is also a static `newBuilder` method, which return value must NOT be ignored.

Therefore your approach is way too broad. Let's discuss in person if needed.

File: base/src/main/java/io/spine/code/java/Annotations.java [hunk 2]
```diff
@@ -51,10 +56,20 @@ private Annotations() {
...
-     * @return an {@link AnnotationSpec} describing the {@link Generated javax.annotation.Generated}
+     * @return an {@code AnnotationSpec} describing the {@link Generated javax.annotation.Generated}
      *         annotation
      */
     public static AnnotationSpec generatedBySpineModelCompiler() {
         return GENERATED;
     }
+
+    /**
+     * Generates {@code \@CanIgnoreReturnValue} annotation spec.
```
Comment (armiol) [inline base/src/main/java/io/spine/code/java/Annotations.java]: I know another  approach is to use `{@literal @}`
```
Generates {@code {@literal @}CanIgnoreReturnValue} annotation spec.
```
Please investigate which one is better.

Comment (dmitrykuzmin) [reply base/src/main/java/io/spine/code/java/Annotations.java]: It seems that just `{@code @CanIgnoreReturnValue}` is displayed okay in both Idea doc reference and the doc HTML generated by the Gradle.

When using any of the approaches for escaping the literal though, the doc is not displayed properly, so I just removed the escape mechanism here and in the similar case.
